app/code/community/OnePica/AvaTax/Model/Service/Avatax16/Abstract.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * OnePica_AvaTax
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0), a
+ * copy of which is available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   OnePica
+ * @package    OnePica_AvaTax
+ * @author     OnePica Codemaster <codemaster@onepica.com>
+ * @copyright  Copyright (c) 2015 One Pica, Inc.
+ * @license    http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+
+/**
+ * Avatax 16 service abstract model
+ *
+ * @category   OnePica
+ * @package    OnePica_AvaTax
+ * @author     OnePica Codemaster <codemaster@onepica.com>
+ */
+abstract class OnePica_AvaTax_Model_Service_Avatax16_Abstract
+{
+    /**
+     * The module helper
+     *
+     * @var OnePica_AvaTax_Helper_Data
+     */
+    protected $_helper = null;
+
+    /**
+     * The module address helper
+     *
+     * @var OnePica_AvaTax_Helper_Address
+     */
+    protected $_addressHelper = null;
+
+    /**
+     * The module config helper
+     *
+     * @var OnePica_AvaTax_Helper_Config
+     */
+    protected $_configHelper = null;
+
+    /**
+     * The module config helper
+     *
+     * @var OnePica_AvaTax_Helper_Errors
+     */
+    protected $_errorsHelper = null;
+
+    /**
+     * Returns the AvaTax helper.
+     *
+     * @return OnePica_AvaTax_Helper_Data
+     */
+    protected function _getHelper()
+    {
+        if (!$this->_helper) {
```

Comment:
No sense to do this caching. return Mage::helper is enough. Please replace in all places.